### PR TITLE
Add zip files to releases for Windows

### PR DESCRIPTION
**Pull request by bufdev**
This PR creates a zip file for Windows releases that includes the following:

```
buf/
buf/LICENSE
buf/bin/
buf/bin/buf.exe
buf/bin/protoc-gen-buf-breaking.exe
buf/bin/protoc-gen-buf-lint.exe
```

Fixes https://github.com/bufbuild/buf/issues/3166

### CODE REVIEW

File: make/buf/scripts/release.bash [hunk 1]
```diff
@@ -138,6 +140,20 @@ for os in Darwin Linux; do
...
 
+for os in Windows; do
+  for arch in x86_64 arm64; do
+    dir="${os}/${arch}/${BASE_NAME}"
+    # "${os}/${arch}"
+    zip_context_dir="$(dirname "${dir}")"
+    zip_dir="${BASE_NAME}"
+    zipfile="${BASE_NAME}-${os}-${arch}.zip"
+    pushd "${zip_context_dir}" >/dev/null
+    zip -r "${zipfile}" "${zip_dir}"
```
Comment (oliversun9) [inline make/buf/scripts/release.bash:151]: I ran `SKIP_MINISIGN=1 make bufrelease` and it gives me the error that `zip` command is not found (in the [docker image](https://github.com/bufbuild/buf/blob/0b436b4cf494dd93cfc0ddb446d4fab5b07b514c/make/buf/all.mk#L157)). I also tried `golang:1.22-bullseye` and `golang:1.22-alpine` but neither worked.

**Comment by oliversun9** [comment]
With [zip installed](https://github.com/bufbuild/buf/pull/3175), the assets now look like:

```
ls .build/release/buf/assets                    
buf-Darwin-arm64                           protoc-gen-buf-breaking-Darwin-x86_64
buf-Darwin-arm64.tar.gz                    protoc-gen-buf-breaking-Linux-aarch64
buf-Darwin-x86_64                          protoc-gen-buf-breaking-Linux-x86_64
buf-Darwin-x86_64.tar.gz                   protoc-gen-buf-breaking-Windows-arm64.exe
buf-Linux-aarch64                          protoc-gen-buf-breaking-Windows-x86_64.exe
buf-Linux-aarch64.tar.gz                   protoc-gen-buf-lint-Darwin-arm64
buf-Linux-x86_64                           protoc-gen-buf-lint-Darwin-x86_64
buf-Linux-x86_64.tar.gz                    protoc-gen-buf-lint-Linux-aarch64
buf-Windows-arm64.exe                      protoc-gen-buf-lint-Linux-x86_64
buf-Windows-arm64.zip                      protoc-gen-buf-lint-Windows-arm64.exe
buf-Windows-x86_64.exe                     protoc-gen-buf-lint-Windows-x86_64.exe
buf-Windows-x86_64.zip                     sha256.txt
protoc-gen-buf-breaking-Darwin-arm64
```

and unzipping one of them:

```
unzip -l .build/release/buf/assets/buf-Windows-x86_64.zip
Archive:  .build/release/buf/assets/buf-Windows-x86_64.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  07-22-2024 14:45   buf/
    11357  07-22-2024 14:45   buf/LICENSE
        0  07-22-2024 14:44   buf/bin/
 35875840  07-22-2024 14:44   buf/bin/buf.exe
 18874368  07-22-2024 14:44   buf/bin/protoc-gen-buf-breaking.exe
 12030464  07-22-2024 14:44   buf/bin/protoc-gen-buf-lint.exe
---------                     -------
 66792029                     6 files
 ```

**Review by oliversun9 — APPROVED** [review]

**Review by oliversun9 — APPROVED** [review]
Approved once https://github.com/bufbuild/buf/pull/3175 is merged

**Review by doriable — APPROVED** [review]

**Review by doriable — APPROVED** [review]

**Comment by bufdev** [comment]
LGTM

**Pull request merged**